storybook/src/stories/UseArray/Burger/Burger.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useState } from 'react'
+import { useArray } from 'too-many-hooks'
+import { Button, Select } from '../../../components'
+import '../../../index.css'
+import {
+  INGREDIENTS,
+  INGREDIENT_GROUPS,
+  INGREDIENT_GROUP_OPTIONS,
+  INGREDIENT_OPTIONS,
+} from './constants'
+import { Ingredient, IngredientGroup } from './types'
+
+const Burger: React.FC = () => {
```

Comment:
The burger does a good job of demonstrating a simple stack. Too much beyond that has diminishing returns, imo.

How about a bookshelf? Filtering by "genre" is pretty intuitive. So is inserting at arbitrary locations, as opposed to a bun in the middle of a burger. Plus, the indices should be easier to display without clutter since they would flow horizontally. Or they could just not be there at all since it should be easier to count the discrete elements.

Reply:
A to-do list could work as well. I'm having trouble thinking of a case for the replacement utils 🤔

Reply:
Yeah, I think that the functions are useful but just not usually in a way that is very good for direct interaction. Since the filter updates the state permanently it is hard to think of a use case for standard user interactions since generally filters are always removable. Perhaps that specific function just doesn't need its own story.

I do like the bookshelf for insertion/removal though, I will get started on that.

Reply:
Sweet. Yeah I agree that there are probably times where something might be too niche for a story, and maybe also if the functionality is obvious enough.

It brings up an interesting point though. Do we want to refrain from adding features that we can't think of a use case for?

As it relates to the array, there are methods that already return a modified version of the array, `map` and `filter` for example, that the developer can use in conjunction with `set` to update the array state. Others like inserting make more sense for us to wrap, since it's already awkward enough to do this in an immutable way and there's more value in making it easier.

Reply:
A solution to this might be an extensible approach to the functionality our hooks provide, as in making sure to expose fundamental composable tools that the developer can use to solve more sophisticated problems instead of packing in too much unnecessarily.

In the filter case, returning `set` is fine since if the developer wants to permanently filter the array, they can just do 

```ts
const [array, { set }] = useArray([1, 2, 3])
set(array.filter(callback))
```
 instead of

```ts
const [array, { filter }] = useArray([1, 2, 3])
filter(callback)
```

Reply:
I don't think that the issue is that it doesn't have a use, just that it doesn't have a use that is something usually directly triggered by an end-user, so hard to write a story for. I don't think not having a story for a specific function of a hook is indicative of it not being useful. Exposing the `filter` to allow the more elegant version of updating costs basically nothing and can help simplify code. Perhaps something to help clarify would be renaming the function to something like `removeWhere` to indicate that it is a stateful operation, not a standard filter. I was torn between going with the common JS name and a more permanent-state-update-specific name.

Also, it is currently being used in the `Burger` story so it definitely can be included in a story, it just needs to be one where it makes sense to do a permanent update like removing a certain type of ingredient from a burger. I think we can keep it in use there and remove some of the other functionality in favor of the bookshelf to simplify the story.